### PR TITLE
ENH: pad: normalize `pad_width` on the host for torch

### PR DESCRIPTION
**Pull request by rgommers**
`pad_width` is host metadata (an int or (before, after) pairs), but the torch delegation branch normalized it by round-tripping through a tensor: `xp.asarray(pad_width)` places it on the *default* device, and the `tuple(...)` call to build the python ints that `torch.nn.functional.pad` expects then performs one device-to-host transfer per element. With a CUDA default device that is a needless host-to-device copy plus 2*ndim synchronizing `.item()` reads per call; with a data-free `meta` default device it raises outright.

Factor the pure-python normalization out of `_funcs.pad` into `normalize_pad_width` and reuse it in the torch branch, reversing the axis order on the host.

This also aligns the torch branch with the documented `pad_width` types (the tensor broadcast incidentally accepted undocumented forms such as shape ``(n, 1)``).

Found this when working on non-default device handling issues in `scipy`; SciPy tests will cover this so I won't worry about introducing multi-device testing here right now.

AI Generation Disclosure: patch generated with Claude Fable 5 (had to tweak both code comments and commit message a bit, but the hard work was done by Claude here).

### CODE REVIEW

File: src/array_api_extra/_lib/_funcs.py [hunk 1]
```diff
@@ -538,6 +538,22 @@ def nunique(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     )
 
 
+def normalize_pad_width(
```
Comment (lucascolley) [inline src/array_api_extra/_lib/_funcs.py]: Arguably this belongs in `_helpers` instead

File: src/array_api_extra/_delegation.py [hunk 2]
```diff
@@ -792,12 +791,13 @@ def pad(
...
     if is_torch_namespace(xp):
-        pad_width = xp.asarray(pad_width)
-        pad_width = xp.broadcast_to(pad_width, (x.ndim, 2))
-        pad_width = xp.flip(pad_width, axis=(0,)).flatten()
-        return xp.nn.functional.pad(x, tuple(pad_width), value=constant_values)  # type: ignore[arg-type]  # pyright: ignore[reportArgumentType]
+        # normalize `pad_width` on the host rather than through a tensor as done in
+        # `torch/_numpy`'s implementation (avoids device transfers)
+        pad_width_seq = _funcs.normalize_pad_width(pad_width, x.ndim)
+        # torch.nn.functional.pad counts dimensions from the last one
+        flat_pad_width = [w for pair in reversed(pad_width_seq) for w in pair]
```
Comment (lucascolley) [inline src/array_api_extra/_delegation.py]: non-blocking but something like this makes this expression easier to parse to my eyes
```suggestion
        flat_pad_width = [
            w
            for pair in reversed(pad_width_seq)
            for w in pair
        ]
```

**Comment by lucascolley** [comment]
> SciPy tests will cover this so I won't worry about introducing multi-device testing here right now.

Would be good to at least open an issue for this

**Comment by rgommers** [comment]
It's already covered by the generic issue for non-default devices; it's one of very many (see gh-25567) so not worth a separate issue.

File: src/array_api_extra/_delegation.py (hunk 2 above)
Comment (rgommers) [reply src/array_api_extra/_delegation.py]: sure, let's apply that change.

**Comment by lucascolley** [comment]
> It's already covered by the generic issue for non-default devices; it's one of very many (see gh-25567) so not worth a separate issue. 

Sorry, I meant an issue on this repo

File: src/array_api_extra/_lib/_funcs.py (hunk 1 above)
Comment (rgommers) [reply src/array_api_extra/_lib/_funcs.py]: I'll move it if you prefer that - but are you sure? It's specific to just `pad` and not useful as a standalone generic helper, so keeping it right above `def pad:` would be my choice.

**Comment by rgommers** [comment]
Ah, that seems reasonable - a generic "introduce multi-device or non-default device testing for array-api-extra" issue. I think after we have it for SciPy, it might be clearer if/how it can be done in this repo.

File: src/array_api_extra/_lib/_funcs.py (hunk 1 above)
Comment (lucascolley) [reply src/array_api_extra/_lib/_funcs.py]: Given that it is used in 2 separate modules, I would rather move it to `_helpers`, yes. Let's add it to `__all__` over there, too.

**Review by lucascolley — APPROVED** [review]

**Comment by rgommers** [comment]
Hmm, looks like `torch` isn't tested in CI, since only the linter failed.

**Comment by lucascolley** [comment]
> Hmm, looks like `torch` isn't tested in CI, since only the linter failed.

it is tested otherwise coverage would fail... I think the previous commit still worked due to the helper being imported into `_funcs`. That is what the error code suggests at least: `reportPrivateLocalImportUsage`

**Comment by lucascolley** [comment]
thanks Ralf!
